components/shared/embedder/lib.rs

```diff
@@ -223,6 +225,18 @@ pub enum AllowOrDeny {
     Deny,
 }
 
+/// Data about a `WebView` of `<iframe>` viewport: its size and also the
+/// HiDPI scale factor to use when rendering the contents.
+#[derive(Clone, Copy, Debug, Default, Deserialize, MallocSizeOf, PartialEq, Serialize)]
+pub struct ViewportDetails {
+    /// The size of the layout viewport.
+    pub size: Size2D<f32, CSSPixel>,
```

Comment:
`WindowSizeData`'s doc comment includes a spec link to https://www.w3.org/TR/css-viewport/#initial-viewport

Is that not relevant for this new struct?

Reply:
Yeah. This isn't highly relevant any longer. This data structure is used for many more things now than just setting the css-viewport properties. I think the implementation of this CSS idea just moved gradually further and further from this data structure through the years. [This field](https://github.com/servo/stylo/blob/127952e9cc0d881c08ac5c41b626eb38674432e9/style/servo/media_queries.rs#L57) in Stylo is a better representation of where `initial-viewport` is handled.